leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/tlv/LwM2mNodeTlvEncoder.java

```diff
@@ -66,6 +69,8 @@ public byte[] encode(LwM2mNode node, LwM2mPath path, LwM2mModel model, LwM2mValu
     }
 
     private static class InternalEncoder implements LwM2mNodeVisitor {
+    
+        private LinkSerializer linkSerializer = new DefaultLinkSerializer();
```

Comment:
should be moved at LwM2mNodeTlvEncoder level with a custom constructor

Reply:
done, but InternalEncoder is not static any more (to access the non static parent field)